okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.kt

```diff
@@ -283,37 +289,34 @@ class HttpLoggingInterceptor @JvmOverloads constructor(
     logger.log(headers.name(i) + ": " + value)
   }
 
-  companion object {
-    /**
-     * Returns true if the body in question probably contains human readable text. Uses a small
-     * sample of code points to detect unicode control characters commonly used in binary file
-     * signatures.
-     */
-    internal fun Buffer.isUtf8(): Boolean {
-      try {
-        val prefix = Buffer()
-        val byteCount = size.coerceAtMost(64)
-        copyTo(prefix, 0, byteCount)
-        for (i in 0 until 16) {
-          if (prefix.exhausted()) {
-            break
-          }
-          val codePoint = prefix.readUtf8CodePoint()
-          if (Character.isISOControl(codePoint) && !Character.isWhitespace(codePoint)) {
-            return false
-          }
+  /**
+   * Returns true if the body in question probably contains human readable text. Uses a small
+   * sample of code points to detect unicode control characters commonly used in binary file
+   * signatures.
+   */
+  private fun Buffer.isUtf8(): Boolean {
+    try {
+      val prefix = Buffer()
+      val byteCount = size.coerceAtMost(64)
+      copyTo(prefix, 0, byteCount)
+      for (i in 0 until 16) {
```

Comment:
Can be `repeat(16) {`.

Reply:
Unfortunately that doesn’t support `break`.